board_helpers: support sof_card_private initialization

brentlu opened the pull request:
Move common board quirks to sof_board_helper.h then implement an helper function sof_intel_board_get_ctx() for machine drivers to parse these board quirks and setup sof_card_private structure accordingly.

brentlu committed:
ASoC: Intel: board_helpers: support sof_card_private initialization

Add a helper function for machine drivers to initialize common part of
sof_card_private structure. Also unify the macros of board quirks for
the initialization.

Signed-off-by: Brent Lu <brent.lu@intel.com>

brentlu committed:
ASoC: Intel: sof_cs42l42: use common module for sof_card_private initialization

Use intel_board module to initialize sof_card_private structure.

Signed-off-by: Brent Lu <brent.lu@intel.com>

brentlu committed:
ASoC: Intel: sof_nau8825: use common module for sof_card_private initialization

Use intel_board module to initialize sof_card_private structure.

Signed-off-by: Brent Lu <brent.lu@intel.com>

brentlu committed:
ASoC: Intel: sof_rt5682: use common module for sof_card_private initialization

Use intel_board module to initialize sof_card_private structure

Signed-off-by: Brent Lu <brent.lu@intel.com>

brentlu committed:
ASoC: Intel: sof_ssp_amp: use common module for sof_card_private initialization

Use intel_board module to initialize sof_card_private structure.

Signed-off-by: Brent Lu <brent.lu@intel.com>

plbossart commented on sound/soc/intel/boards/sof_cs42l42.c:
```diff
@@ -306,7 +306,7 @@ static const struct platform_device_id board_ids[] = {
 				SOF_CS42L42_SSP_AMP(1) |
 				SOF_CS42L42_NUM_HDMIDEV(4) |
-				SOF_BT_OFFLOAD_PRESENT |
+				SOF_CS42L42_BT_OFFLOAD_PRESENT |
```
separate patch, this had nothing to do with the context stuff.

brentlu replied:
fixed

brentlu committed:
ASoC: Intel: sof_cs42l42: rename BT offload quirk

Rename the quirk in preparation for future changes: common quriks will
be defined and handled in board helper module.

Signed-off-by: Brent Lu <brent.lu@intel.com>

brentlu commented:
@bardliao Would you check if this PR is good enough for you? Thanks.

brentlu commented:
@plbossart Shall we merge this PR? Thanks.

plbossart commented:
@bardliao I'll let you merge
